.github/workflows/ci-image-scanning.yaml

```diff
@@ -1,10 +1,13 @@
 name: image-scanning
 on:
   push: 
+  pull_request:
 jobs:
   use-trivy-to-scan-image:
     name: image-scanning
-    if: ${{ github.repository == 'karmada-io/karmada' }}
+    # Trigger by pull_request to work for dependabot, for running github/codeql-action/upload-sarif
+    # Otherwise trigger by push
+    if: ( github.repository == 'karmada-io/karmada' && github.actor == 'dependabot[bot]' && github.event_name == 'pull_request') || ( github.repository == 'karmada-io/karmada' && github.actor != 'dependabot[bot]' && github.event_name == 'push' ) 

```

Comment:
Feels a little bit tricky, is it the recommended way?

Reply:
I understand that expressions is look a little scary, and we have other two option:

1. only trigger this workflow on push:

```diff
on:
--  push: 
++  pull_request:
```

2. run with two job, the first job working for push and the second job working for dependabot & pull_request

```yaml
on:
  push:
  pull_request:
  
jobs:
  use-trivy-to-scan-image:
    name: image-scanning
    if:  github.repository == 'karmada-io/karmada' && github.actor == 'dependabot[bot]' && github.event_name == 'pull_request'

  use-trivy-to-scan-image-bot:
  if: github.repository == 'karmada-io/karmada' && github.actor != 'dependabot[bot]' && github.event_name == 'push'
...
```

I would suggest option 1, which is simple enough and meets the requirements.

Reply:
> only trigger this workflow on push:

Do you mean on `pull_request`?

Now the workflow is exactly [on-push](https://github.com/karmada-io/karmada/blob/c310f89c126db93fd722447c21962201b071f1e3/.github/workflows/ci-image-scanning.yaml#L3).